src/VisualStudio/Core/Def/Progression/GraphProvider.cs

```diff
@@ -176,10 +176,13 @@ public IEnumerable<GraphCommand> GetCommands(IEnumerable<GraphNode> nodes)
     {
         EnsureInitialized();
 
+        // Gather most information about the nodes in a single pass.
+        var nodesInfo = GetTypeKindAndSymbolKindInfo(nodes);
+
         // Only nodes that explicitly state that they contain children (e.g., source files) and named types should
         // be expandable.
-        if (nodes.Any(n => n.Properties.Any(p => p.Key == DgmlNodeProperties.ContainsChildren)) ||
-            nodes.Any(n => IsAnySymbolKind(n, SymbolKind.NamedType)))
+        if (nodesInfo.ContainsSymbolKindNamedType ||
+            nodes.Any(static n => n.Properties.Any(static p => p.Key == DgmlNodeProperties.ContainsChildren)))
```

Comment:
😕 Can you not just use `(bool?)n[DgmlNodeProperties.ContainsChildren] ?? false` or something like that? Currently this code appears to assume that if the key exists, its value is true. I'm not sure that's valid.

Reply:
Maybe? I'm not sure as the semantics around properties is a bit odd (the GetValues call for properties filters properties that are valid based on the owner's schema). Either way, unless we're certain the current behavior isn't correct, I'd rather not change the behavior.